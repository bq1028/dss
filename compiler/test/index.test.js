const dss = require('../')

describe('dss', () => {
  it('compiles', async () => {
    const src = `
      .a { color: red; }
      .a:hover { color: blue; }
      @media screen and (min-width: 30px) {
        .a { color: hotpink }
      }
    `
    const styles = await dss(src)

    expect(styles).toMatchSnapshot()
    expect(src + '\n\n⬇⬇⬇⬇\n\n' + dss.flush()).toMatchSnapshot()
  })

  it('compiles fallbacks', async () => {
    const src = `
      .a {
        color: red;
        color: green;
      }
    `
    const styles = await dss(src)

    expect(styles).toMatchSnapshot()
    expect(src + '\n\n⬇⬇⬇⬇\n\n' + dss.flush()).toMatchSnapshot()
  })
})
