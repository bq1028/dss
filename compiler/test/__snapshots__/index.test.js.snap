// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dss compiles 1`] = `
Object {
  "a": Array [
    "dss_rfc3hq-169mlyl",
    "dss_16bwgo5-11z5xnj",
    "dss_1u3exn4-nfznl2",
  ],
}
`;

exports[`dss compiles 2`] = `
"
      .a { color: red; }
      .a:hover { color: blue; }
      @media screen and (min-width: 30px) {
        .a { color: hotpink }
      }
    

⬇⬇⬇⬇

.dss_1u3exn4-nfznl2:hover{color:blue}.dss_rfc3hq-169mlyl{color:red}@media screen and (min-width: 30px){.dss_16bwgo5-11z5xnj{color:hotpink}}"
`;

exports[`dss compiles fallbacks 1`] = `
Object {
  "a": Array [
    "dss_rfc3hq-1t0ure0",
  ],
}
`;

exports[`dss compiles fallbacks 2`] = `
"
      .a {
        color: red;
        color: green;
      }
    

⬇⬇⬇⬇

.dss_rfc3hq-1t0ure0{color:red;color:green}"
`;

exports[`dss does not have duplicates 1`] = `
Object {
  "bar": Array [
    "dss_14e3233-hlr2nt",
  ],
  "foo": Array [
    "dss_14e3233-hlr2nt",
  ],
}
`;

exports[`dss does not have duplicates 2`] = `".dss_14e3233-hlr2nt{display:block}"`;
